test/e2e/alertmanager_test.go

```diff
@@ -203,6 +206,10 @@ func TestAlertmanagerKubeRbacProxy(t *testing.T) {
 	if !ok {
 		t.Fatalf("couldn't get silenceID from response %q", string(b))
 	}
+	defer func() {
+		clients["editor"].Do("DELETE", fmt.Sprintf("/api/v2/silence/%s", silID), sil)
+		t.Log("deleting silence as 'editor' for cleanup")
+	}()
```

Comment:
This does not seen to work as expected. Even running this test successfully (including the cleanup delete) seems to cause an issue in a repeated run.
```
alertmanager_test.go:243: expecting 1 silence, got 2 ("[{\"id\":\"0a7c6481-53e7-4052-91fa-9535b447de2b\",\"status\":{\"state\":\"active\"},\"updatedAt\":\"2021-09-27T13:47:52.677Z\",\"comment\":\"some comment\",\"createdBy\":\"somebody\",\"endsAt\":\"2021-09-27T16:47:50.000+02:00\",\"matchers\":[{\"isRegex\":false,\"name\":\"namespace\",\"value\":\"test-kube-rbac-proxy\"},{\"isRegex\":false,\"name\":\"alertname\",\"value\":\"Drill\"}],\"startsAt\":\"2021-09-27T13:47:52.677Z\"},{\"id\":\"d6b4e7b5-9263-4e2b-8686-021b104d0485\",\"status\":{\"state\":\"expired\"},\"updatedAt\":\"2021-09-27T13:15:00.940Z\",\"comment\":\"some comment\",\"createdBy\":\"somebody\",\"endsAt\":\"2021-09-27T13:15:00.940Z\",\"matchers\":[{\"isRegex\":false,\"name\":\"namespace\",\"value\":\"test-kube-rbac-proxy\"},{\"isRegex\":false,\"name\":\"alertname\",\"value\":\"Drill\"}],\"startsAt\":\"2021-09-27T13:15:00.108Z\"}]\n")
```
Input welcome!

Reply:
@jan--f `Do` returns a response and an error but we are handling neither. Maybe we should grab the error and see if this delete is working as expected?

Reply:
Yeah I agree, we really should. The funny thing is that it seems like the silence doesn't get deleted on a successful run, in which case this cleanup delete would actually return an error.

```
=== RUN   TestAlertmanagerKubeRbacProxy
    alertmanager_test.go:89: creating namespace "test-kube-rbac-proxy"
    alertmanager_test.go:115: creating service account "editor"
    alertmanager_test.go:122: creating role binding "editor" -> "monitoring-rules-edit"
    alertmanager_test.go:115: creating service account "viewer"
    alertmanager_test.go:122: creating role binding "viewer" -> "monitoring-rules-view"
    alertmanager_test.go:115: creating service account "anonymous"
    alertmanager_test.go:183: creating silence as "viewer" (denied)
    alertmanager_test.go:183: creating silence as "anonymous" (denied)
    alertmanager_test.go:192: creating silence as 'editor' (allowed)
    alertmanager_test.go:215: listing silences as 'anonymous' (denied)
    alertmanager_test.go:224: listing silences as "viewer" (allowed)
    alertmanager_test.go:224: listing silences as "editor" (allowed)
    alertmanager_test.go:276: deleting silence as "viewer" (denied)
    alertmanager_test.go:276: deleting silence as "anonymous" (denied)
    alertmanager_test.go:285: deleting silence as 'editor' (allowed)
    alertmanager_test.go:211: deleting silence as 'editor' for cleanup
```

The test runs to completion without an error. Then on re-run I get

```
=== RUN   TestAlertmanagerKubeRbacProxy
    alertmanager_test.go:89: creating namespace "test-kube-rbac-proxy"
    alertmanager_test.go:115: creating service account "viewer"
    alertmanager_test.go:122: creating role binding "viewer" -> "monitoring-rules-view"
    alertmanager_test.go:115: creating service account "anonymous"
    alertmanager_test.go:115: creating service account "editor"
    alertmanager_test.go:122: creating role binding "editor" -> "monitoring-rules-edit"
    alertmanager_test.go:183: creating silence as "viewer" (denied)
    alertmanager_test.go:183: creating silence as "anonymous" (denied)
    alertmanager_test.go:192: creating silence as 'editor' (allowed)
    alertmanager_test.go:215: listing silences as 'anonymous' (denied)
    alertmanager_test.go:224: listing silences as "viewer" (allowed)
    alertmanager_test.go:243: expecting 1 silence, got 2 ("[{\"id\":\"0a7c6481-53e7-4052-91fa-9535b447de2b\",\"status\":{\"state\":\"active\"},\"updatedAt\":\"2021-09-27T13:47:52.677Z\",\"comment\":\"some comment\",\"createdBy\":\"somebody\",\"endsAt\":\"2021-09-27T16:47:50.000+02:00\",\"matchers\":[{\"isRegex\":false,\"name\":\"namespace\",\"value\":\"test-kube-rbac-proxy\"},{\"isRegex\":false,\"name\":\"alertname\",\"value\":\"Drill\"}],\"startsAt\":\"2021-09-27T13:47:52.677Z\"},{\"id\":\"d6b4e7b5-9263-4e2b-8686-021b104d0485\",\"status\":{\"state\":\"expired\"},\"updatedAt\":\"2021-09-27T13:15:00.940Z\",\"comment\":\"some comment\",\"createdBy\":\"somebody\",\"endsAt\":\"2021-09-27T13:15:00.940Z\",\"matchers\":[{\"isRegex\":false,\"name\":\"namespace\",\"value\":\"test-kube-rbac-proxy\"},{\"isRegex\":false,\"name\":\"alertname\",\"value\":\"Drill\"}],\"startsAt\":\"2021-09-27T13:15:00.108Z\"}]\n")
    alertmanager_test.go:211: deleting silence as 'editor' for cleanup
    ```